### PR TITLE
Decoupling: Use camelCase for API responses in the editor

### PR DESCRIPTION
**Pull request by mohdsayed**
## Context

A fresh version of https://github.com/google/web-stories-wp/pull/9683

## Summary

As discussed, we are ignoring to camel case the following objects ( not the key itself )
- image `sizes` ( Not `mime_type` and `source_url` in `sizes` object. )
- `capabilities`, `extra_capabilities`
- `story_data`
- `visibility` ( in `getTaxonomies` )
- `meta`

Following items have been updated to camel case

- [x] permalink_template
- [x] story_data
- [x] style_presets
- [x] preview_link
- [x] edit_link
- [x] embed_post_link
- [x] featured_media
- [x] featured_media_src
- [x] publisher_logo
- [x] date_gmt
- [x] modified_gmt
- [x] web_stories_is_muted
- [x] length_formatted
- [x] file_size
- [x] file_name
- [x] lock_user
- [x] Taxonomy Labels
```js
- search_items
- all_items
- parent_item
- parent_item_colon
- edit_item
- view_item
- update_item
- add_new_item
- new_item_name
- separate_items_with_commas
- add_or_remove_items
- choose_from_most_used
- not_found
- no_terms
- filter_by_item
- items_list_navigation
- items_list
- most_used
- back_to_items
- item_link
- item_link_description
- menu_name
- name_admin_bar
```

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses [#456](https://github.com/google/web-stories-wp/issues/2918)

### CODE REVIEW

File: packages/wp-story-editor/src/api/media.js [hunk 1]
```diff
@@ -188,7 +188,7 @@ export function updateMedia(config, mediaId, data) {
     path: `${config.api.media}${mediaId}/`,
     data,
     method: 'POST',
-  });
+  }).then(getResourceFromAttachment);
```
Comment (swissspidy) [inline packages/wp-story-editor/src/api/media.js:191]: Should we update `deleteMedia` below to return `void`?

See previous discussion at https://github.com/google/web-stories-wp/pull/9683/files#r758728852

File: packages/wp-story-editor/src/api/pageTemplate.js [hunk 2]
```diff
@@ -64,5 +65,5 @@ export function deletePageTemplate(config, id) {
     }),
     data: { force: true },
     method: 'POST',
-  });
+  }); // Response is not being used in core editor.
```
Comment (swissspidy) [inline packages/wp-story-editor/src/api/pageTemplate.js:114]: Let's make it return `void`, otherwise if it gets used at some point it will be confusing.

**Comment by swissspidy** [comment]
> As discussed, we are ignoring to camel case the following objects ( not the key itself )
> 
> * image `sizes` ( including `mime_type` and `source_url` in `sizes` object. )

To clarify, I'm only opposed to camelCasing image sizes _**names**_ like `web_stories_thumbnail`. But we should definitely camelCase `sourceUrl` and the like. #9683 had this (`v0037_camelCaseResourceSizes.js`)

**Comment by mohdsayed** [comment]
@swissspidy 

> camelCasing image sizes names **like** `web_stories_thumbnail`

Do you mean, we should camel case sizes like `medium_large` and `post-thumbnail` and only ignore key names like `web_stories_*` or `web-stories_*`?



image

**Comment by swissspidy** [comment]
> @swissspidy
> 
> > camelCasing image sizes names **like** `web_stories_thumbnail`
> 
> Do you mean, we should camel case sizes like `medium_large` and `post-thumbnail` and only ignore key names like `web_stories_*` or `web-stories_*`?
> 
> image

No, I mean ignore image size names but camelCase sourceUrl

**Comment by mohdsayed** [comment]
@swissspidy Created another PR for media sizes https://github.com/google/web-stories-wp/pull/10013

**Review by swissspidy — APPROVED** [review]

**Pull request merged**